tests/conftest.py

```diff
@@ -42,6 +42,25 @@ def fast_dump():
         yield chunked_read(f, 500)
 
 
+@pytest_asyncio.fixture
+def fast_dump_with_extra_header_params():
+    """
+    Add header parameters to `fast_dump.bin` binary stream. The fixture
+    is used for testing absolute timing parameters passed in the header.
+    """
+    def fast_dump(extra_header_params):
+        param_to_replace = "sample_bytes=\"52\""
+        params = " ".join(f"{k}=\"{v}\"" for k, v in extra_header_params.items())
+        params = " ".join([param_to_replace, params]) if params else param_to_replace
+        with open(Path(__file__).parent / "data_dumps/fast_dump.bin", "rb") as f:
+            # Simulate larger chunked read
+            data = chunked_read(f, 500)
+            for buffer in data:
+                yield buffer.replace(param_to_replace.encode(), params.encode())
+
+    return fast_dump
+
+
```

Comment:
I think this will do for now. In future it might be nice to have a script to generate the binaries for tests so we can easily add new fields.